stl/inc/__msvc_int128.hpp

```diff
@@ -0,0 +1,1155 @@
+
+        _STL_INTERNAL_CHECK(__q[2] == 0);
+        return static_cast<uint64_t>(__q[1]) << 32 | __q[0];
+    }
+
+    constexpr _Uint128() noexcept : _Word{} {}
+
+    constexpr _Uint128(const integral auto _Val) noexcept : _Word{static_cast<uint64_t>(_Val)} {
+        if constexpr (signed_integral<decltype(_Val)>) {
+            if (_Val < 0) {
+                _Word[1] = ~0ull;
+            }
+        }
+    }
+
+    explicit constexpr _Uint128(const uint64_t _Low, const uint64_t _High) noexcept : _Word{_Low, _High} {}
+
+    template <integral _Ty>
+    _NODISCARD explicit constexpr operator _Ty() const noexcept {
+        return static_cast<_Ty>(_Word[0]);
+    }
+
+    _NODISCARD explicit constexpr operator bool() const noexcept {
+        return (_Word[0] | _Word[1]) != 0;
+    }
+
+    _NODISCARD constexpr friend bool operator==(const _Uint128& _Left, const _Uint128& _Right) noexcept {
+        return _Left._Word[0] == _Right._Word[0] && _Left._Word[1] == _Right._Word[1];
+    }
+
+    _NODISCARD constexpr friend strong_ordering operator<=>(const _Uint128& _Left, const _Uint128& _Right) noexcept {
+        strong_ordering _Ord = _Left._Word[1] <=> _Right._Word[1];
+        if (_Ord == strong_ordering::equal) {
+            _Ord = _Left._Word[0] <=> _Right._Word[0];
+        }
+        return _Ord;
+    }
+
+    _NODISCARD friend constexpr _Uint128 operator<<(const _Uint128& _Left, const _Uint128& _Right) noexcept {
+        auto _Tmp{_Left};
+        _Tmp._Left_shift(static_cast<unsigned char>(_Right._Word[0]));
+        return _Tmp;
+    }
+
+    constexpr _Uint128& operator<<=(const _Uint128& _Count) noexcept {
+        _Left_shift(static_cast<unsigned char>(_Count._Word[0]));
+        return *this;
+    }
+    template <integral _Ty>
+    friend constexpr _Ty& operator<<=(_Ty& _Left, const _Uint128& _Right) noexcept {
+        _Left = static_cast<_Ty>(_Uint128{_Left} << _Right);
+        return _Left;
+    }
+
+    _NODISCARD friend constexpr _Uint128 operator>>(const _Uint128& _Left, const _Uint128& _Right) noexcept {
+        auto _Tmp{_Left};
+        _Tmp._Right_shift(static_cast<unsigned char>(_Right));
+        return _Tmp;
+    }
+
+    constexpr _Uint128& operator>>=(const _Uint128& _Count) noexcept {
+        _Right_shift(static_cast<unsigned char>(_Count._Word[0]));
+        return *this;
+    }
+    template <integral _Ty>
+    friend constexpr _Ty& operator>>=(_Ty& _Left, const _Uint128& _Right) noexcept {
+        _Left = static_cast<_Ty>(_Uint128{_Left} >> _Right);
+        return _Left;
+    }
+
+    constexpr _Uint128& operator++() noexcept {
+        if (++_Word[0] == 0) {
+            ++_Word[1];
+        }
+        return *this;
+    }
+    constexpr _Uint128 operator++(int) noexcept {
+        auto _Tmp = *this;
+        ++*this;
+        return _Tmp;
+    }
+
+    constexpr _Uint128& operator--() noexcept {
+        if (_Word[0]-- == 0) {
+            --_Word[1];

```

Comment:
`subborrow_u64` might be better (but need to prove that if the change is considered)

Reply:
Adding this to #2520.